pip_audit/_dependency_source/resolvelib/pypi_provider.py

```diff
@@ -188,12 +195,26 @@ def _get_metadata_for_sdist(self) -> Message:
 
         return metadata
 
+    def _generate_dist_hashes(self, dist_data: bytes) -> None:
+        """
+        Populate a mapping of hash algorithm names to hashes for the candidate.
+        """
+        if self.name not in self.req_hashes:
+            return
+        hash_algorithms = self.req_hashes[self.name].keys()
+        dist_hashes = {}
+        for alg in hash_algorithms:
+            hasher = hashlib.new(alg, dist_data)
```

Comment:
Small thing, but `hashlib.new` actually has subtle behavior we don't necessarily want: it'll dynamically look up *any* digest functions provided by the underlying OpenSSL library, even ones that aren't explicitly supported by `hashlib` itself.

To work around that, we probably want to check `hash_algorithms` against `hashlib.algorithms_guaranteed` first, and warn/error on any identifiers that we don't recognize.

Reply:
Good catch. Thanks.

Reply:
[911cd0d](https://github.com/pypa/pip-audit/pull/462/commits/911cd0d2cde1c439de3f5cef43216feeae49d3e9)